near-sdk/src/store/bucket/iter.rs

```diff
@@ -0,0 +1,210 @@
+use std::iter::FusedIterator;
+
+use borsh::{BorshDeserialize, BorshSerialize};
+
+use super::{Bucket, Container, ERR_INCONSISTENT_STATE};
+use crate::{env, store::vec};
+
+impl<'a, T> IntoIterator for &'a Bucket<T>
+where
+    T: BorshSerialize + BorshDeserialize,
+{
+    type Item = &'a T;
+    type IntoIter = Iter<'a, T>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.iter()
+    }
+}
+
+impl<'a, T> IntoIterator for &'a mut Bucket<T>
+where
+    T: BorshSerialize + BorshDeserialize,
+{
+    type Item = &'a mut T;
+    type IntoIter = IterMut<'a, T>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.iter_mut()
+    }
+}
```

Comment:
This looks peculiar. Seems to also be duplicated in `near-sdk/src/store/bucket/impls.rs` as well, or is this some kind of weird requirement somewhere?

Reply:
oh nice catch haha, the impls module wasn't included in the parent module. Removed and added